src/Servers/IIS/AspNetCoreModuleV2/CommonLib/HostFxr.cpp

```diff
@@ -99,3 +107,61 @@ HostFxrErrorRedirector HostFxr::RedirectOutput(RedirectionOutput* writer) const
 {
     return HostFxrErrorRedirector(m_corehost_set_error_writer_fn, writer);
 }
+
+int HostFxr::InitializeForApp(int argc, const PCWSTR* argv, const std::wstring dotnetExe, bool callStartupHook) const noexcept
+{
+    if (m_hostfxr_initialize_for_app_fn == nullptr)
+    {
+        return 0;
+    }
+
+    hostfxr_initialize_parameters params;
+    params.size = sizeof(hostfxr_initialize_parameters);
+    params.host_path = L"";
+
+    if (!dotnetExe.empty())
+    {
+        std::filesystem::path dotnetExePath(dotnetExe);
+
+        auto dotnetFolderPath = dotnetExePath.parent_path();
+        params.dotnet_root = dotnetFolderPath.c_str();
+        RETURN_IF_NOT_ZERO(m_hostfxr_initialize_for_app_fn(argc - 1, &argv[1], nullptr, &params, &m_host_context_handle));
+
+    }
+    else
+    {
+        params.dotnet_root = L"";
+
+        // Initialize_for_app doesn't work with an exe name
+        std::filesystem::path applicationPath(argv[0]);
+        applicationPath.replace_extension(".dll");
```

Comment:
Work around for https://github.com/dotnet/core-setup/issues/6388 for now.

Reply:
Actually, I'd rather say we are blocked on it. There are a few other scenarios that still cause failing tests. StartsWithWebConfigVariationsPortable tests a bunch of different parameter configurations and 2 of them are failing. I can again work around it, but at this point it's getting quite bad.